src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -175,6 +204,140 @@ impl ScalarFunc {
         }
         Ok(Some(i64::from(LittleEndian::read_u32(&input[0..4]))))
     }
+
+    pub fn aes_encrypt<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let crypt = try_opt!(self.children[0].eval_string(ctx, row));
+        let key = try_opt!(self.children[1].eval_string(ctx, row));
+        let mode = match ctx.aes_mode {
+            Some(mode) => mode,
+            None => {
+                ctx.warnings
+                    .append_warning(Error::warn_option_ignored("aes mode"));
+                return Ok(None);
+            }
+        };
+
+        if mode.iv_required && self.children.len() == 3 {
+            ctx.warnings
+                .append_warning(Error::warn_option_ignored("IV"));
+        }
+        let key_size = mode.key_size;
+        let aes_key_size = match get_key_size(key_size) {
+            Ok(size) => size,
+            Err(err) => {
+                ctx.warnings.append_warning(err);
+                return Ok(None);
+            }
+        };
+        let key_mysql = derive_key_mysql(key.as_ref(), key_size);
+
+        match mode.mode_name {
+            "ecb" => {
+                let mut encryptor =
+                    aes::ecb_encryptor(aes_key_size, key_mysql.as_ref(), blockmodes::PkcsPadding);
+                let mut final_result = Vec::<u8>::new();
+                let mut read_buffer = buffer::RefReadBuffer::new(crypt.as_ref());
+                let mut buffer = [0; 4096];
+                let mut write_buffer = buffer::RefWriteBuffer::new(&mut buffer);
+                loop {
```

Comment:
Is it possible to use [this interface](https://github.com/DaGenix/rust-crypto/blob/cc1a5fde1ce957bd1a8a2e30169443cdb4780111/src/symmetriccipher.rs#L51), which seems to be simpler?